### ISSUE TITLE
Unexpected token: punc (.)

### ISSUE THREAD

**Issue by banderon1** [sensedeep/dynamodb-onetable#305]
When trying to build my typescript (version: "4.6.2") project, I receive the following error:
```
ERROR in index.js from Terser
Unexpected token: punc (.) [index.js:28938,32]
```

A quick google search led me to this StackOverflow response:
https://stackoverflow.com/a/70897123/693807

Which mentions that terser has issues with optional chaining.
The line that is referenced in my output is [here](https://github.com/sensedeep/dynamodb-onetable/blob/main/src/Table.js#L937):
```
let client = params?.client || this.client
```
I noticed that this line recently changed (according to this issue: https://github.com/sensedeep/dynamodb-onetable/issues/293), and is likely the culprit to my error.

I'm using version: "2.1099.0" of the `aws-sdk`, and version: "2.3.4" of `dynamodb-onetable`. I'm not sure how the `V3` variable is loaded or used, but it _should_ be falsey.

**To Reproduce**

Here's the entirety of my schema file:
```
export default {
  format: 'onetable:1.1.0',
  version: '0.0.1',
  indexes: {
    primary: { hash: 'pk', sort: 'sk' },
    gsi1: { hash: 'sk', sort: 'pk', follow: true },
  },
  models: {
    Meeting: {
      pk: { type: String, value: 'MEETING#${id}' },
      sk: { type: String, value: 'EMAIL#${email}' },
      id: { type: String, generate: 'ulid' },
      name: { type: String },
      status: { type: String, default: 'created', enum: ['created', 'started', 'confirmed', 'verified', 'canceled', 'ended', 'deleted'] },
      type: { type: String, default: 'Meeting' },
    },
    User: {
      pk: { type: String, value: 'USER#${userName}' },
      sk: { type: String, value: 'EMAIL#${email}' },
      userName: { type: String, required: true },
      email: { type: String, required: true },
      firstName: { type: String },
      lastName: { type: String },
      type: { type: String, default: 'User' },
    }
  } as const,
  params: {
    isoDates: true,
    timestamps: true,
  },
};
```

```
ERROR in index.js from Terser
Unexpected token: punc (.) [index.js:28938,32]
```

**Expected behavior**
Build passes successfully. If I remove all references to the `V3` variable, then it does succeed.

**Environment (please complete the following information):**
 - OS: OSX 12.2.1
 - Node Version: v16.14.0
 - OneTable Version: 2.3.4
 - TypeScript Version: 4.6.2
 - AWS SDK Version: 2.1099.0

**Comment by mobsense** [comment]
Does upgrading to a more recent version of Terser work?  Typescript can parse that syntax.

**Comment by mobsense** [comment]
This is due to terser not parsing that syntax which is now part of Javascript. TSC does parse it.

Here is a quick patch:

We'll revert this in the next release.

```
diff --git a/src/Table.js b/src/Table.js
index 3f991a1..530bc92 100644
--- a/src/Table.js
+++ b/src/Table.js
@@ -934,7 +934,7 @@ export class Table {
     */
     unmarshall(item, params) {
         if (this.V3) {
-            let client = params?.client || this.client
+            let client = params.client ? params.client : this.client
             let options = client.params.unmarshall
             if (Array.isArray(item)) {
                 for (let i = 0; i < item.length; i++) {
@@ -957,7 +957,7 @@ export class Table {
     }

     marshallv2(item, params) {
-        let client = params?.client || this.client
+        let client = params.client ? params.client : this.client
         for (let [key, value] of Object.entries(item)) {
             if (value instanceof Set) {
                 item[key] = client.createSet(Array.from(value))
```

**Comment by mobsense** [comment]
Fixed in 2.3.5
